modules/nextflow/src/main/groovy/nextflow/executor/TaskArrayExecutor.groovy

```diff
@@ -54,4 +57,31 @@ interface TaskArrayExecutor {
      */
     String getArrayTaskId(String jobId, int index)
 
+    default boolean isWorkDirDefaultFS() {
+        getWorkDir().fileSystem== FileSystems.default
+    }
+
+    /**
+     * Get a {@link TaskHandler} work directory for the task array resolution
+     *
+     * @param handler
+     * @return
+     */
+    default String getArrayWorkDir(TaskHandler handler) {
+        return isFusionEnabled()
+            ? FusionHelper.toContainerMount(handler.task.workDir).toString()
+            : handler.task.workDir.toUriString()
+    }
+
+    default String getArrayLaunchCommand(String taskDir) {
+        if( isFusionEnabled() ) {
+            return "bash ${taskDir}/${TaskRun.CMD_RUN}"
+        }
+        else if( isWorkDirDefaultFS() ) {
+            return "bash ${taskDir}/${TaskRun.CMD_RUN} 2>&1 > ${taskDir}/${TaskRun.CMD_LOG}"
```

Comment:
```suggestion
            return "bash ${taskDir}/${TaskRun.CMD_RUN} 2>&1 | tee ${taskDir}/${TaskRun.CMD_LOG}"
```

I believe this should use `tee` so that the log output is also forwarded to stdout like regular tasks

Reply:
But then the parent job will collect the output of child tasks 

https://github.com/nextflow-io/nextflow/blob/037a67d9e5cedadd5302107cb76c1f55f0f940d7/modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy#L72-L72

Reply:
Actually I guess we should disable this option for the job array, it makes no sense to send anything to the log file in the parent directory

Reply:
Then we'll still want to use `tee` so that the task log is also the job stdout. Some schedulers like PBS Pro can display the stdout of a job while it's running, but you need to either print to stdout or set the `-o` option. We can't do the latter because it's a job array but we can do the former, each child job will print its log to its own stdout

Reply:
I'm worried funneling the stdout of all tasks to the same log. Read this 

> Note: if you only use '%A' in the log all array tasks will try to write to a single file. The performance of the run will approach zero asymptotically. Make sure to use both %A and %a in the log file name specification.

https://help.rc.ufl.edu/doc/Submitting_Array_Jobs

Reply:
But the `taskDir` in this method refers to the child task dir, so each child job will send its stdout to its own log and also to its own stdout

Also, the note you mentioned won't happen anyway if we disable the `-o` option when submitting a job array

Reply:
Well, I'm not 100% certain that all grid executors handle the job array stdout in this way. Might be better to play it safe for now, only sending the child job stdout to its log. But either way we should remove the `-o` option for job arrays, it can only cause trouble

Reply:
If we disabled the `-o` option in the parent job then `tee` should not be used otherwise it won't be consumed by anything.

Reply:
> But either way we should remove the -o option for job arrays, it can only cause trouble

Make sense to remove it. Let's do in the parent PR